### PR TITLE
Add .nfo ratings tag

### LINKED ISSUE #4887: No movie/tv show rating if library scan only local nfo (no metadownloader)
There is no movie/tv show rating if the library scan using only local nfo file (Kodi format)

**System (please complete the following information):**
 - OS: Windows 10
 - Jellyfin Version: 10.6.2

**To Reproduce**

1. A movie/tv show with nfo file contains information as Kodi format (https://kodi.wiki/view/NFO_files/Movies).
2. There is rating field in nfo file.
3. In library setting, remove all Movie metadata downloaders (means use local info only).
4. Scan library.

**Expected behavior**
New added movie/tv show should have rating info

**Screenshots**

**Additional context**
With my limited knowledge, I guess the problem seems the parser assume it is a simple xml tag value while in fact it is multiple complex sub tags.

From MediaBrowser.XbmcMetadata\Parsers\BaseNfoParser.cs (assume Jellyfin use this to scan local nfo file - I'm unsure)
```csharp
                case "rating":
                    {
                        var rating = reader.ReadElementContentAsString();

                        if (!string.IsNullOrWhiteSpace(rating))
                        {
                            // All external meta is saving this as '.' for decimal I believe...but just to be sure
                            if (float.TryParse(rating.Replace(',', '.'), NumberStyles.AllowDecimalPoint, CultureInfo.InvariantCulture, out var val))
                            {
                                item.CommunityRating = val;
                            }
                        }

                        break;
                    }
```

Real life example of rating tags:
```xml
  <ratings>
    <rating default="true" max="10" name="imdb">
      <value>7.7</value>
      <votes>260935</votes>
    </rating>
  </ratings>
```

### PR DESCRIPTION
**Pull request by Ullmie02**
**Changes**
Add support for the `<ratings>` tag in .nfo files.
Because of compatibility reasons I didn't remove support for the `<rating>` tag.

Ratings with names containing `tomato` will be treated as `CriticRatings`, every other rating will get imported as `CommunityRating`

**Issues**
Fixes #4887

### CODE REVIEW

File: MediaBrowser.XbmcMetadata/Parsers/BaseNfoParser.cs [hunk 1]
```diff
@@ -866,6 +881,90 @@ private void FetchFromVideoNode(XmlReader reader, T item)
...
+                    {
+                        case "value":
+                            var val = reader.ReadElementContentAsString();
+
+                            if (!string.IsNullOrWhiteSpace(val))
+                            {
+                                if (float.TryParse(val.Replace(',', '.'), NumberStyles.AllowDecimalPoint, CultureInfo.InvariantCulture, out var ratingValue))
+                                {
+                                    // if ratingName contains tomato --> assume critic rating
+                                    if (ratingName != null && ratingName.Contains("tomato", StringComparison.InvariantCultureIgnoreCase))
```
Comment (Bond-009) [inline MediaBrowser.XbmcMetadata/Parsers/BaseNfoParser.cs:944]:
```suggestion
                                    if (ratingName != null && ratingName.Contains("tomato", StringComparison.OrdinalIgnoreCase))
```

File: MediaBrowser.XbmcMetadata/Parsers/BaseNfoParser.cs [hunk 2]
```diff
@@ -866,6 +881,90 @@ private void FetchFromVideoNode(XmlReader reader, T item)
...
+                if (reader.NodeType == XmlNodeType.Element)
+                {
+                    switch (reader.Name)
+                    {
+                        case "value":
+                            var val = reader.ReadElementContentAsString();
+
+                            if (!string.IsNullOrWhiteSpace(val))
+                            {
+                                if (float.TryParse(val.Replace(',', '.'), NumberStyles.AllowDecimalPoint, CultureInfo.InvariantCulture, out var ratingValue))
```
Comment (Bond-009) [inline MediaBrowser.XbmcMetadata/Parsers/BaseNfoParser.cs:941]: I don't like this hack

**Review by Bond-009 — APPROVED** [review]

File: MediaBrowser.XbmcMetadata/Parsers/BaseNfoParser.cs (hunk 2 above)
Comment (Ullmie02) [reply MediaBrowser.XbmcMetadata/Parsers/BaseNfoParser.cs]: This hack was used for the `rating` tag so I copied it over. Other number tags don't use it so I guess it's fine removing it

File: MediaBrowser.XbmcMetadata/Parsers/BaseNfoParser.cs [hunk 3]
```diff
@@ -866,6 +881,90 @@ private void FetchFromVideoNode(XmlReader reader, T item)
...
+                    {
+                        case "value":
+                            var val = reader.ReadElementContentAsString();
+
+                            if (!string.IsNullOrWhiteSpace(val))
+                            {
+                                if (float.TryParse(val, NumberStyles.AllowDecimalPoint, CultureInfo.InvariantCulture, out var ratingValue))
+                                {
+                                    // if ratingName contains tomato --> assume critic rating
+                                    if (ratingName != null && ratingName.Contains("tomato", StringComparison.OrdinalIgnoreCase))
```
Comment (Bond-009) [inline MediaBrowser.XbmcMetadata/Parsers/BaseNfoParser.cs:944]: This doesn't handle
```xml
		<rating name="tomatometerallcritics" max="10">
			<value>7.6</value>
			<votes>71</votes>
		</rating>
		<rating name="tomatometerallaudience" max="10">
			<value>6.2</value>
			<votes>119873</votes>
		</rating>
```
Correclty

**Review by Bond-009 — CHANGES_REQUESTED** [review]

File: MediaBrowser.XbmcMetadata/Parsers/BaseNfoParser.cs (hunk 3 above)
Comment (Ullmie02) [reply MediaBrowser.XbmcMetadata/Parsers/BaseNfoParser.cs:944]: So additional check if it doesn't contain `audience`?

**Comment by Uatschitchun** [comment]
Would so love to be able to have the ratings, which are all present in nfo being read by jellyfin ;)

**Review by Bond-009 — APPROVED** [review]
Still feels like a bit of a hack :p

**Review by crobibero — APPROVED** [review]

**Pull request merged**